students/csdotson/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Lesson 4 - Mailroom... part 2
+
+### Function Definitions ###
+def prompt(prompt, menu):
+    # Create menu based on arguments
+    while True:
+        response = input(prompt)
+        if menu[response]() == "quit":
+            break
+
+
+def send_thank_you():
+    # Create sub_prompt for creating thank you's and adding new donors/donations
+    prompt(thank_you_prompt, thank_you_menu)
+
+
+def add_donation():
+    # Append donation to existing donor, or add donor/donation to existing list
+    response = input("\nPlease enter a donor name: ")
+    if response not in donations.keys():
+        new_donor = response
+        new_donation = input(f"Please enter a donation amount for {new_donor}: ")
+        donations[new_donor] = [float(new_donation)]
+        print_email(new_donor, new_donation)
+    else:
+         current_donor = response
+         new_donation = input(f"Please enter a donation amount for {current_donor}: ")
+         donations[current_donor].append(float(new_donation))
+         print_email(current_donor, new_donation)
```

Comment:
you can take the print_email out of if/else block since its duplicated

also, no need to rename `response` var, you can call it `donor_entry`.. so basically to simplify:
```
donor_name_entry = input("Please enter donor name")
donation = float(input(f"Please enter a donation amount for {donor_name_entry}: "))
if donor_entry in donations:  # no need to call keys()
    donations[donor_name_entry] = [donation]
else:
    donations[donor_name_entry].append(donation)
print_email(donor_name_entry, donation)
```

you can even make it simpler if you use defaultdict